### DISCUSSION TITLE
Orthotropic elasticity in @Implicitgenericbehaviour

### DISCUSSION THREAD

**Discussion by yy74774** [thelfer/tfel#469] [General]
Hi!
I am a beginner in _mfront_ and would like to ask you about the code in the case of _implicitgenericbehavior_.
I would like to use _mfront_ to describe the transversely isotropic thermo-poro-mechanical elasto-plastic behavior of Callovo-Oxfordian claystone.

I have written an isotropic elastoplastic model that considers mechanical and temperature loads (in the case of _implicitgenericbehavior_), and it has proven to be very effective!
Next I would like to extend this model to the anisotropic case, and I would like to start with anisotropic elasticity.

In the _implicit_ case, I successfully modeled using @ComputeStiffnessTensor. But when I use _implicitgenericbehavior_, this function does not work, so I calculated the items of the stiffness matrix myself and assembled them. This time _mfront_ can create the material behavior, but _mtest_ does not complete the calculation.
In the attachment you can find me my current _mfront_ and _mtest_ code.
[Orthotropic_elasticity.zip](https://github.com/thelfer/tfel/files/13917245/Orthotropic_elasticity.zip)

For this case, I have the following questions:
1. Is it possible to modify my current code to achieve anisotropic elasticity under _implicitgenericbehavior_?
2. Is it possible to consider pore mechanics in the _implicit_ case, if this is feasible I think I can do the full material mechanics behavior in the _implicit_ case.

Thanks in advance,
Junfeng

**Comment by thelfer** [comment]
Dear Junfeng,

Thanks for your interest for MFront.

You may want to use directly the `computeOrthotropicStiffnessTensor` which is defined in the `TFEL/Material/Stiffness.hxx` header. I rapidly modified your code to do so, you'll have to adapt the definition of the elastic properties.

I also corrected a mistake in `@TangentOperator` (you wrote `De = De;`).

Thomas

[ela_ani_generic.zip](https://github.com/thelfer/tfel/files/13917572/ela_ani_generic.zip)
